go/def.bzl

```diff
@@ -592,6 +621,53 @@ go_binary = rule(
 
 go_test = rule(
     go_test_impl,
+    attrs = go_library_attrs + go_tool_attrs + _crosstool_attrs + {
+        "test_generator": attr.label(
+            executable = True,
+            default = Label(
+                "//go/tools:generate_test_main",
+            ),
+            cfg = "host",
+        ),
+        "x_defs": attr.string_dict(),
+    },
+    executable = True,
+    fragments = ["cpp"],
+    outputs = {
+        "lib": "%{name}.a",
+        "main_lib": "%{name}_main_test.a",
+        "main_go": "%{name}_main_test.go",
+    },
+    test = True,
+)
+
+# go_tool_* rules are for use building tools required by BUILD rules.
```

Comment:
I don't think we need these rules. You should be able to add filter_tags to the set of tools we build with "go build" inside go_repositories. See _go_repositories_tools_impl in go/private/go_repositories.bzl. fetch_repo and gazelle are already built this way.